src/backend/distributed/relay/relay_event_utility.c

```diff
@@ -755,8 +756,10 @@ AppendShardIdToName(char **name, uint64 shardId)
 	neededBytes = snprintf((*name), NAMEDATALEN, "%s", extendedName);
 	if (neededBytes < 0)
 	{
+		char *strerrno = strerror(errno);
+
 		ereport(ERROR, (errcode(ERRCODE_OUT_OF_MEMORY),
-						errmsg("out of memory: %s", strerror(errno))));
+						errmsg("out of memory: %s", strerrno)));
```

Comment:
We can use `%m` here.
```suggestion
						errmsg("out of memory: %m")));
```